### PR TITLE
fix(core): Fix worker shutdown errors when active executions

### PR DESCRIPTION
**Pull request by ivov**
https://linear.app/n8n/issue/PAY-1817

### CODE REVIEW

File: packages/cli/src/commands/worker.ts [hunk 1]
```diff
@@ -61,23 +61,6 @@ export class Worker extends BaseCommand {
 
 		try {
 			await this.externalHooks?.run('n8n.stop', []);
-
-			const hardStopTimeMs = Date.now() + this.gracefulShutdownTimeoutInS * 1000;
```
Comment (ivov) [inline packages/cli/src/commands/worker.ts:65]: Note that this time was misreporting, the graceful shutdown had already started at `BaseCommand.onTerminationSignal`

**Review by netroy — APPROVED** [review]
🙇🏽

File: packages/cli/src/commands/BaseCommand.ts [hunk 2]
```diff
@@ -333,7 +333,9 @@ export abstract class BaseCommand extends Command {
 			this.logger.info(`Received ${signal}. Shutting down...`);
 			this.shutdownService.shutdown();
 
-			await Promise.all([this.stopProcess(), this.shutdownService.waitForShutdown()]);
+			await this.shutdownService.waitForShutdown();
+
+			await this.stopProcess();
```
Comment (tomi) [inline packages/cli/src/commands/BaseCommand.ts:338]: This also changes the shutdown behaviour of the `start` command. It seems to have some duplication (some functions are called in both `ShutdownService` and `StartCommand.stopProcess()`. We should fix that also as a follow-up

**Review by tomi — APPROVED** [review]
Thank you for addressing this 🙇

**Comment by janober** [comment]
Got released with [`n8n@1.55.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.55.0)

**Pull request merged**